test/compress/template-string.js

```diff
@@ -0,0 +1,48 @@
+template_strings: {
+    input: {
+        ``;
+        `xx\`x`;
+        `${ foo + 2 }`;
+        ` foo ${ bar + `baz ${ qux }` }`;
+    }
+    expect_exact: "``;`xx\\`x`;`${foo+2}`;` foo ${bar+`baz ${qux}`}`;";
+}
+
+template_string_prefixes: {
+    input: {
+        String.raw`foo`;
+        foo `bar`;
+    }
+    expect_exact: "String.raw`foo`;foo`bar`;";
+}
+
+template_strings_ascii_only: {
+    beautify = {ascii_only: true}
+    input: {
+        var foo = `foo
+        bar
+        ↂωↂ`
+    }
+    expect_exact: "var foo=`foo\\n        bar\\n        \\u2182\\u03c9\\u2182`;"
+}
+
+template_strings_without_ascii_only: {
+    input: {
+        var foo = `foo
+        bar
+        ↂωↂ`
+    }
+    expect_exact: "var foo=`foo\n        bar\n        ↂωↂ`;"
+}
+
+template_string_with_constants: {
+    options = {
+        evaluate: true
+    }
+    input: {
+        var foo = `${4 + 4} equals 4 + 4`;
```

Comment:
That's done I guess? (I've done quite some extreme testing somewhere else, but there could be more examples)